### PR TITLE
Added support to callflow modules and pivot to utilize custom KVs

### PR DESCRIPTION
**Pull request by danielfinke**
Long lost follow up to both #1328 and #1255.

Addressed concerns about cross-app dependencies by moving custom-KV-related code into kapps_call.
Moved to a generic variable replacer in cf_exe.
Schema doc is empty template since variables (the properties of module's data) are freeform.

### CODE REVIEW

File: applications/callflow/src/cf_exe.erl [hunk 1]
```diff
@@ -650,7 +650,9 @@ launch_cf_module(#state{call=Call
                        ,cf_module_pid=OldPidRef
                        }=State) ->
     Module = <<"cf_", (kz_json:get_value(<<"module">>, Flow))/binary>>,
-    Data = kz_json:get_value(<<"data">>, Flow, kz_json:new()),
+    Data = apply_dynamic_values(kz_json:get_value(<<"variables">>, Flow, kz_json:new())
```
Comment (lazedo) [inline applications/callflow/src/cf_exe.erl]: `variables` should be added to callflow schema

File: core/kazoo_json/src/kz_json.erl [hunk 2]
```diff
@@ -236,6 +237,23 @@ are_identical(JObj1, JObj2) ->
 from_list([]) -> new();
 from_list(L) when is_list(L) -> ?JSON_WRAPPER(L).
 
+-spec recursive_from_list(json_proplist() | json_array()) -> object() | json_array().
+-spec recursive_from_list(json_proplist() | json_array(), json_proplist()) ->
+                                 json_proplist() | json_array().
+recursive_from_list([]) -> from_list([]);
+recursive_from_list(L) when is_list(L) ->
```
Comment (fenollp) [inline core/kazoo_json/src/kz_json.erl]: Could you add a couple test cases to kz_json_test?

**Comment by danielfinke** [comment]
@fenollp tests added, fixed an issue discovered in those tests. Rebased to latest master

**Comment by jamesaimonetti** [comment]
@danielfinke I pulled this down locally, rebased, and dialyzed a bit. See the branch [here](https://github.com/2600hz/kazoo/tree/voxter-master-pr-custom-kvs) and let me know if this is sufficient for your purposes. I'll merge that branch and reference this PR

**Comment by danielfinke** [comment]
@jamesaimonetti your branch looks solid. It'll work. Thanks!

**Comment by fenollp** [comment]
@jamesaimonetti Planning on taking action here?

**Comment by jamesaimonetti** [comment]
Hey @danielfinke, wow did we lose the thread on this one! I've rebased and updated my [branch](https://github.com/2600hz/kazoo/tree/voxter-master-pr-custom-kvs) based of this one. Mind having a look and testing it out, make sure it still is doing what you expected?

**Comment by jamesaimonetti** [comment]
Closing in favor of #4892; move discussion there

**Pull request closed without merging**